cmd/mt-parrot/generate.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"time"
+
+	"github.com/grafana/metrictank/schema"
+)
+
+func produceArtificialMetrics(schemas []*schema.MetricData) {
+	for tick := range time.NewTicker(artificialMetricsInterval).C {
+		for _, metric := range schemas {
+			metric.Time = tick.Unix()
+			metric.Value = float64(tick.Unix())
+		}
+		gateway.Flush(schemas)
+		log.Infof("flushed schemas for ts %d", tick.Unix())
+	}
+}
+
+//generateSchemas generates a MetricData that hashes to each of numPartitions partitions
+func generateSchemas(numPartitions int32) []*schema.MetricData {
+	var metrics []*schema.MetricData
+	for i := int32(0); i < numPartitions; i++ {
+		metrics = append(metrics, generateSchema(i))
+	}
+	return metrics
+}
+
+//generateSchema generates a single MetricData that hashes to the given partition
+func generateSchema(desiredPartition int32) *schema.MetricData {
+	metric := schema.MetricData{
+		OrgId:    orgId,
+		Unit:     "partyparrots",
+		Mtype:    "gauge",
+		Interval: int(artificialMetricsInterval.Seconds()),
+	}
+
+	for i := 1; true; i++ {
+		metric.Name = fmt.Sprintf("parrot.testdata.%d.generated.%s", desiredPartition, generatePartitionSuffix(i))
+		id, err := metric.PartitionID(partitionMethod, partitionCount)
+		if err != nil {
+			log.Fatal(err)
+		}
+		if id == desiredPartition {
+			log.Infof("metric for partition %d: %s", desiredPartition, metric.Name)
+			return &metric
+		}
+	}
+	return nil
+}
+
+var alphabet = []rune("abcdefghijklmnopqrstuvwxyz")
+
+//generatePartitionSuffix deterministically generates a suffix for partition by brute force
+func generatePartitionSuffix(i int) string {
+	if i == 0 {
+		return ""
+	}
+	return generatePartitionSuffix(i/26) + string(alphabet[i%26])
```

Comment:
a few small issues here:
1) empty string is invalid
2) you have an off by one error (26/26 should use the 1st element which has index 0, this uses index 1 which is the 2nd element)

I think this will work better:
```
func generatePartitionSuffix(i int) string {
        if i > 25 {
                return generatePartitionSuffix((i/26)-1) + string(alphabet[i%26])
        }
        return string(alphabet[i%26])
}
```

Reply:
good catch, a8d3295f6a564442a1becef93f69a87dbead6a39